src/pages-and-resources/discussions/app-config-form/apps/shared/blackout-dates/BlackoutDatesInput.jsx

```diff
@@ -0,0 +1,90 @@
+import React, { useState } from 'react';
+import { Form, TransitionReplace } from '@edx/paragon';
+import { useFormikContext } from 'formik';
+import PropTypes from 'prop-types';
+
+import { checkFieldErrors } from '../../../utils';
+
+const BlackoutDatesInput = ({
+  index,
+  value,
+  type,
+  label,
+  fieldName,
+  helpText,
+  fieldClasses,
+  helperClasses,
+  formGroupClasses,
+}) => {
+  const {
+    handleChange, handleBlur, errors, touched,
+  } = useFormikContext();
+  const [inFocus, setInFocus] = useState(false);
+  const isInvalidInput = checkFieldErrors(touched, errors, 'blackoutDates', fieldName, index);
+
+  const renderFormFeedback = (message, messageType = 'default') => (
+    <Form.Control.Feedback type={messageType} hasIcon={false}>
+      <div className={`small ${helperClasses}`}>{message}</div>
+    </Form.Control.Feedback>
+  );
+
+  const handleFocusOut = (event) => {
+    handleBlur(event);
+    setInFocus(false);
+  };
+
+  return (
+    <Form.Group
+      controlId={`blackoutDates.${index}.${fieldName}`}
+      className={`col ${formGroupClasses}`}
+      isInvalid={!inFocus && isInvalidInput}
+    >
+      <Form.Control
+        value={value}
+        type={type}
+        onChange={handleChange}
+        floatingLabel={label}
+        className={fieldClasses}
+        onBlur={(event) => handleFocusOut(event)}
+        onFocus={() => setInFocus(true)}
+      />
+      <TransitionReplace key={index} className="mt-1">
+        {inFocus ? (
+          <React.Fragment key="open">{renderFormFeedback(helpText)}</React.Fragment>
+        ) : (
+          <React.Fragment key="closed" />
+        )}
+      </TransitionReplace>
+      <TransitionReplace key={`${index}-error`}>
+        {!inFocus && isInvalidInput ? (
+          <React.Fragment key="open">
```

Comment:
is it same as `<>`?

Reply:
It is the same but <> does bot support the key attribute.